codegen/mockgen.go

```diff
@@ -22,48 +22,35 @@ package codegen
 
 import (
 	"bytes"
+	"github.com/golang/mock/mockgen/model"
+	"github.com/pkg/errors"
 	"go/token"
 	"os"
 	"os/exec"
 	"path"
-	"path/filepath"
 	"runtime"
 	"sort"
 	"strconv"
 	"strings"
-
-	"github.com/golang/mock/mockgen/model"
-	"github.com/pkg/errors"
 )
 
 const (
-	exampleGatewayPkg = "github.com/uber/zanzibar/examples/example-gateway"
-	mockgenPkg        = "github.com/golang/mock/mockgen"
-	zanzibarPkg       = "github.com/uber/zanzibar"
+	mockgenPkg = "github.com/golang/mock/mockgen"
 )
 
 // MockgenBin is a struct abstracts the mockgen binary built from mockgen package in vendor
 type MockgenBin struct {
 	// Bin is the absolute path to the mockgen binary built from vendor
 	Bin string
 
-	// projRoot is the absolute path of the project, it is also where the vendor directory is
-	projRoot  string
 	pkgHelper *PackageHelper
 	tmpl      *Template
 }
 
 // NewMockgenBin builds the mockgen binary from vendor directory
 func NewMockgenBin(h *PackageHelper, t *Template) (*MockgenBin, error) {
-	pkgRoot := h.PackageRoot()
-	// we would not need this if example gateway is a fully standalone application
-	if pkgRoot == exampleGatewayPkg {
-		pkgRoot = zanzibarPkg
-	}
-	projRoot := filepath.Join(os.Getenv("GOPATH"), "src", pkgRoot)
-
 	// we assume that the vendor directory is flattened as Glide does
-	mockgenDir := path.Join(projRoot, "vendor", mockgenPkg)
+	mockgenDir := path.Join("vendor", mockgenPkg)
```

Comment:
This will end up with a path under `vendor/` which does not exist.

Reply:
That's the intention, I think it's incorrect that it doesn't exist (see jenkins).